SmartDeviceLink/SDLIAPTransport.m

```diff
@@ -247,6 +273,7 @@ - (void)onSessionInitializationCompleteForSession:(SDLIAPSession *)session {
     // Data Session Opened
     if (![controlProtocolString isEqualToString:session.protocol]) {
         self.sessionSetupInProgress = NO;
+        [self sdl_backgroundTaskEnd];
```

Comment:
Per my last comment, I would greatly appreciate understanding why we would not end the background task upon successful connection of the data session.